### DISCUSSION TITLE
More newb friendly Docs/Guides?

### DISCUSSION THREAD

**Discussion by excitedbox** [fomantic/Fomantic-UI#1879] [General]
As someone who comes from a php and DB background a lot of the nodejs stuff is very confusing and I think it would be good if there was more labeling of what is actually being done and the result in the documentation. Right now it only shows the output not the JS that is creating the output.

An example is that when I click view code it shows me the CSS but not the compiled code I would need to upload and WHERE to put it to achieve the result.

The current documentation leaves many questions for newbs and idiots like myself :D mostly in the API usage. Such as project setup with a very basic view. A newbie is usually not trying to implement a bunch of integrations and an expert only needs an API reference of the included functions and files. 

I guess really just more step by step guides assuming the person has never used anything beyond html/css/js/mysql/etc. The big problem people tend to have with documentation is that EVERYONE assumes previous knowledge so NOBODY provides it in a clear concise manner. Especially the people writing frameworks tend to be on the higher skill level so they assume their users are on the same level or more experienced than what they are. 

I am sure most people who can look at the current Docs and understand them without headaches have enough experience that they don't need the docs. It might help to make a little kitchen sink project that shows how to apply the api to the ui and their own code.

I know you inherited the documentation and I am glad to see you are continuing the work and I would be glad to help in any way I can once I get my head wrapped around this project.

**Some questions beginners often have:**
Will the bracketed values get replaced only in the compiled code or can you use it in manually created files?
Can you use LESS code in a live site (ie. printing a string) or will it always need to go through a build?
What if I don't want to use package managers/compiling? (basically applies to the above question)
What bootstrapping code is needed when using just the CSS and Jquery with a DB/Json/API?
Where would the below code go? (or any of these functions ie. pages, or in component.js files or the fomantic js files )

`$.fn.api.settings.api = {
  'get followers' : '/followers/{id}?results={count}',
  'create user'   : '/create',
  'add user'      : '/add/{id}',
  'follow user'   : '/follow/{id}',
  'search'        : '/search/?query={value}'
};`

**Comment by lubber-de** [comment]
Good points! Yes, there is much room for improvement to the documentation. Thanks for the honest opinion 👍 

I think, i can already answer some of your questions, you might not have found them in the current docs:

> An example is that when I click view code it shows me the CSS but not the compiled code I would need to upload and WHERE to put it to achieve the result.

Yes, the docs assume you have basic knowledge of HTML/JS. But we already think of enhancing the examples by implementing a jsfiddle like feature  like flems.io for live examples to play with. https://github.com/fomantic/Fomantic-UI-Docs/issues/218

> I would be glad to help in any way I can once I get my head wrapped around this project.

You are always welcome to help out improving the documentation by providing PRs to https://github.com/fomantic/Fomantic-UI-Docs 

> Will the bracketed values get replaced only in the compiled code or can you use it in manually created files?

Yes, their intention is to be replaced at runtime by their respective data fields [passing data](https://fomantic-ui.com/behaviors/api.html#passing-data)

> Can you use LESS code in a live site (ie. printing a string) or will it always need to go through a build?

 LESS is not parseable natively by the browser, but, as LESS is written in Javascript, you can use LESSJS to compile to CSS in the browser at runtime . More info at http://lesscss.org/

> What if I don't want to use package managers/compiling? (basically applies to the above question)

You can either download / use the precompiled package if you only want to use the default theme or use a CDN. Then you don't hav to deal with nodejs or LESS at all ;)

https://github.com/fomantic/Fomantic-UI-CSS
https://fomantic-ui.com/introduction/getting-started.html#using-a-cdn-provider

> Where would the below code go? (or any of these functions ie. pages, or in component.js files or the fomantic js files )

The example codes are always meant to be used in your own project code to configure or override global functions
